test/hooks.js

```diff
@@ -0,0 +1,682 @@
+'use strict'
+
+const { test } = require('tap')
+const Fastify = require('fastify')
+const { parse, GraphQLSchema } = require('graphql')
+const GQL = require('..')
+
+const sleep = (ms) => new Promise((resolve) => setTimeout(resolve, ms))
+
+process.removeAllListeners('warning')
+
+const schema = `
+  type Query {
+    add(x: Int, y: Int): Int
+  }
+`
+
+const resolvers = {
+  Query: {
+    add: async (_, obj) => {
+      const { x, y } = obj
+      return x + y
+    }
+  }
+}
+
+const query = '{ add(x: 2, y: 2) }'
+
+async function createTestServer (t, customResolvers = resolvers) {
+  const app = Fastify()
+  t.tearDown(() => {
+    app.close()
+  })
+
+  app.register(GQL, { schema, resolvers: customResolvers })
+
+  // needed so that graphql is defined
+  await app.ready()
```

Comment:
This is actually not needed anymore, just returning app would do.

Reply:
Nice, that's good to know!